src/acars_router_servers/listener_servers.rs

```diff
@@ -8,157 +8,150 @@
 
 use crate::generics::OutputServerConfig;
-use crate::helper_functions::should_start_service;
 use crate::tcp_listener_server::TCPListenerServer;
 use crate::tcp_receiver_server::TCPReceiverServer;
 use crate::udp_listener_server::UDPListenerServer;
 use crate::zmq_listener_server::ZMQListnerServer;
-use log::{debug, error, info};
+use serde_json::Value;
 use tokio::sync::mpsc::Sender;
 
 pub fn start_listener_servers(
     config: OutputServerConfig,
-    tx_receivers: Sender<serde_json::Value>,
-    server_type: String,
+    tx_receivers: Sender<Value>,
+    server_type: &str,
 ) {
     // Start the UDP listener servers
 
     // Make sure we have at least one UDP port to listen on
-    if should_start_service(config.listen_udp()) {
+    if let Some(listen_udp) = config.listen_udp {
         // Start the UDP listener servers for server_type
         info!("Starting UDP listener servers for {server_type}");
         start_udp_listener_servers(
-            &server_type,
-            config.listen_udp(),
+            server_type,
+            &listen_udp,
             tx_receivers.clone(),
-            config.reassembly_window().clone(),
+            &config.reassembly_window,
         );
     }
 
     // Start the TCP listeners
 
-    if should_start_service(config.listen_tcp()) {
+    if let Some(listen_tcp) = config.listen_tcp {
         // Start the TCP listener servers for server_type
         info!("Starting TCP listener servers for {server_type}");
         start_tcp_listener_servers(
-            &server_type,
-            config.listen_tcp(),
+            server_type,
+            &listen_tcp,
             tx_receivers.clone(),
-            config.reassembly_window().clone(),
+            &config.reassembly_window,
         );
     }
 
     // Start the ZMQ listeners
 
-    if should_start_service(config.receive_zmq()) {
+    if let Some(receive_zmq) = config.receive_zmq {
         // Start the ZMQ listener servers for {server_type}
         info!("Starting ZMQ Receiver servers for {server_type}");
-        start_zmq_listener_servers(&server_type, config.receive_zmq(), tx_receivers.clone());
+        start_zmq_listener_servers(server_type, &receive_zmq, tx_receivers.clone());
     }
 
-    if should_start_service(config.receive_tcp()) {
+    if let Some(receive_tcp) = config.receive_tcp {
         info!("Starting TCP Receiver servers for {server_type}");
         start_tcp_receiver_servers(
-            &server_type,
-            config.receive_tcp(),
-            tx_receivers.clone(),
-            config.reassembly_window().clone(),
+            server_type,
+            &receive_tcp,
+            tx_receivers,
+            &config.reassembly_window,
         );
     }
 }
 
 fn start_zmq_listener_servers(
-    decoder_type: &String,
-    hosts: &Vec<String>,
-    channel: Sender<serde_json::Value>,
+    decoder_type: &str,
+    hosts: &[String],
+    channel: Sender<Value>,
 ) {
+    let hosts: Vec<String> = hosts.to_vec();
```

Comment:
Because we consume a slice (the `&[String]`), we need to do this to bring it in to scope.

It's a lot more stable than consuming `&Vec<String>`, and allows for the memory to be de-allocated after the call is made.